src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratorDiagnostics.cs

```diff
@@ -21,6 +20,7 @@ public class Ids
             public const string InvalidLibraryImportAttributeUsage = Prefix + "1050";
             public const string TypeNotSupported = Prefix + "1051";
             public const string ConfigurationNotSupported = Prefix + "1052";
+            public const string InterfaceTypeNotSupported = Prefix + "1053";
```

Comment:
Are we expecting other diagnostics around invalid usage of `GeneratedComInterfaceAttribute`? If so, I think it would make sense to have this ID represent invalid usage of the attribute (similar to what we did for `LibraryImport` with `SYSLIB1050`) rather than be specific to `InterfaceType`.